libs/feature/search/src/lib/utils/mapper/elasticsearch.field.mapper.ts

```diff
@@ -199,15 +199,26 @@ export class ElasticsearchFieldMapper {
   }
 
   mapLink(sourceLink: SourceWithUnknownProps): MetadataLink | null {
-    const url = getAsUrl(selectField<string>(sourceLink, 'url'))
+    let url
+    let name
+    let description
+    if (sourceLink.urlObject) {
+      url = sourceLink.urlObject['default'] as string
+      name = sourceLink.nameObject['default']
+      description = sourceLink.descriptionObject['default']
+    } else {
+      url = getAsUrl(selectField<string>(sourceLink, 'url'))
+      name = selectField<string>(sourceLink, 'name')
+      description = selectField<string>(sourceLink, 'description')
+    }
     // no url: fail early
     if (url === null) {
       // TODO: collect errors at the record level?
       console.warn('A link without valid URL was found', sourceLink)
       return null
     }
 
-    const protocolMatch = /^(https?|ftp):/.test(url)
+    const protocolMatch = /^(https?|ftp):/.test(url as string)
```

Comment:
You shouldn't have to specify the type of `url` in various places, that indicates that the original type of this variable hasn't been correctly defined.